lib/acme/client/certificate_request.rb

```diff
@@ -79,7 +79,15 @@ def validate_subject_common_name
 
   def generate
     OpenSSL::X509::Request.new.tap do |csr|
-      csr.public_key = @private_key.public_key
+      if @private_key.is_a?(OpenSSL::PKey::EC) && RbConfig::CONFIG['MAJOR'] == '2' &&
+         RbConfig::CONFIG['MINOR'].to_i < 4
+        # OpenSSL::PKey::EC does not respect classic PKey interface (as defined by
+        # PKey::RSA and PKey::DSA) until ruby 2.4.
+        # Supporting this interface needs monkey patching of OpenSSL:PKey::EC, or
+        # subclassing it. Here, use a subclass.
+        @private_key = ECKeyPatch.new(@private_key)
+      end
+      csr.public_key = @private_key
```

Comment:
What's the reason for not doing `csr.public_key = @private_key.public_key` ?

The `EC` doesn't respond to `public_key`?